src/pool/runner.rs

```diff
@@ -30,6 +30,13 @@ pub trait Runner {
     /// It's guaranteed to be the first method to call before anything else.
     fn start(&mut self, _local: &mut Local<Self::TaskCell>) {}
 
+    /// Called when the runner's thread re-participate in scheduling.
+    ///
+    /// It's used in the scenario of the scale workers. It will treat this
+    /// thread as a new thread. It's guaranteed to be called after start,
+    /// before others else.
+    fn reentrant_start(&mut self, _local: &mut Local<Self::TaskCell>) {}
```

Comment:
Can you update the ASCII graph in the trait definition to show the lifecycle?

Reply:
Fixed, thanks!